apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -65,47 +85,68 @@ def prepare(element_html, data):
             raise Exception("invalid custom format: %s" % custom_format) from None
 
 
-def format_true_ans(element, data, name):
+def format_true_ans(
+    element: lxml.html.HtmlElement, data: pl.QuestionData, name: str
+) -> str:
     a_tru = pl.from_json(data["correct_answers"].get(name, None))
     if a_tru is not None:
         # Get format and comparison parameters
         custom_format = pl.get_string_attrib(element, "custom-format", None)
-        comparison = pl.get_string_attrib(element, "comparison", COMPARISON_DEFAULT)
+        comparison = pl.get_enum_attrib(
+            element, "comparison", ComparisonType, COMPARISON_DEFAULT
+        )
+
         if custom_format is not None:
             a_tru = ("{:" + custom_format + "}").format(a_tru)
-        elif comparison == "relabs":
+        elif comparison is ComparisonType.RELABS:
             # FIXME: render correctly with respect to rtol and atol
             a_tru = "{:.12g}".format(a_tru)
-        elif comparison == "sigfig":
+        elif comparison is ComparisonType.SIGFIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = pl.string_from_number_sigfig(a_tru, digits=digits)
-        elif comparison == "decdig":
+        elif comparison is ComparisonType.DECDIG:
             digits = pl.get_integer_attrib(element, "digits", DIGITS_DEFAULT)
             a_tru = "{:.{ndigits}f}".format(a_tru, ndigits=digits)
         else:
-            raise ValueError(
-                'method of comparison "%s" is not valid (must be "relabs", "sigfig", or "decdig")'
-                % comparison
-            )
+            assert_never(comparison)
     return a_tru
 
 
-def render(element_html, data):
+def get_string_precision(number_string: str) -> float:
+    if "." in number_string:
+        return 10 ** -len(number_string.partition(".")[2])
+    else:
+        return 10 ** (len(number_string) - len(number_string.rstrip("0")))
+
+
+def get_string_significant_digits(number_string: str) -> int:
```

Comment:
It doesn't look like this function gets called anywhere?

Reply:
Not yet; we are only addressing the adding of precision warning for the comparison type `RELABS` in this PR (since this is the scope of #710), but while researching how to do the precision checks I also went ahead and and looked at how to do the significant digits checks (for the other two comparison types). I could remove the code for now and leave this for another PR.

Reply:
Yeah, I think we should remove for now and we can add back later. Shouldn't be too difficult to do in a followup, but would like to avoid any dead code.

Reply:
Sounds good. Removed for now in [faf4997](https://github.com/PrairieLearn/PrairieLearn/pull/7934/commits/faf4997f2b01ae5e2028d597d003ca1a234773d0).